aodupechildimport.php

```diff
@@ -157,6 +157,46 @@ function aodupechildimport_civicrm_pre($op, $objectName, $id, &$params) {
   }
 }
 
+  function aodupechildimport_civicrm_post($op, $objectName, $objectId, &$objectRef) {
+    if ($objectName == "Relationship" && $op == 'create') {
+      // This is done at the time of creating a new relationship between the child and parent.
+      $childRel = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Child of', 'id', 'name_a_b');
+      if ($objectRef->relationship_type_id == $childRel) {
+        createSharedAddress($objectRef->contact_id_b, $objectRef->contact_id_a);
+      }
+    }
+    if ($objectName == 'Individual' && $op == 'edit'
+      && !empty($objectRef->contact_sub_type) && strpos($objectRef->contact_sub_type, 'Child') !== false) {
+      // This is done if the child exists, and is an update. The method above isn't called for existing relationships.
+      $childRel = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Child of', 'id', 'name_a_b');
+      $relationships = civicrm_api3('Relationship', 'get', ['contact_id_a' => $objectId, 'relationship_type_id' => $childRel]);
+      if (!empty($relationships['values'])) {
+        foreach ($relationships['values'] as $relationship) {
+          deleteChildAddresses($relationship['contact_id_a']);
```

Comment:
@Edzelopez should this happen outside of the foreach to handle the situation where parents might be separated?